packages/variants/src/variants.ts

```diff
@@ -6,13 +6,13 @@ function makeThemeFromPartials(
   partialPalette: Partial<IPalette>,
   partialSemantic: Partial<ISemanticColors>
 ): ITheme {
-  return createTheme({
-    ...originalTheme,
-    ...{
-      palette: { ...originalTheme.palette, ...partialPalette },
-      semanticColors: { ...originalTheme.semanticColors, ...partialSemantic }
-    }
-  });
+  // Create variant palette
+  let VariantTheme = createTheme({ palette: { ...originalTheme.palette, ...partialPalette } });
+  // Change semantic colors to use updated variant palette values
+  VariantTheme.semanticColors = { ...VariantTheme.semanticColors, ...partialSemantic };
+  // Fill in the rest of the theme
+  VariantTheme = { ...originalTheme, ...VariantTheme, semanticColors: VariantTheme.semanticColors };
```

Comment:
for this line, take out the ...VariantTheme and just mix in palette and semanticColors directly from VariantTheme
{ ...originalTheme, palette: ...VariantTheme.palette, ...VariantTheme.semanticColors }
this makes sense since the only thing want from VariantTheme is the semanticColors and palette